src/pip/_internal/commands/uninstall.py

```diff
@@ -100,6 +100,8 @@ def run(self, options: Values, args: List[str]) -> int:
                 verbose=self.verbosity > 0,
             )
             if uninstall_pathset:
+                if req.files_skipped:
+                    return ERROR
                 uninstall_pathset.commit()
```

Comment:
I feel the ERROR case still attempt to commit the uninstallation?

Reply:
Or will it return ERROR and stop there? Can you clarify a bit.

Reply:
Basically I’m proposing

```python
uninstall_pathset.commit()
if files_skipped:
    return ERROR
```

Reply:
Done